### DISCUSSION TITLE
custom `_internal` location?

### DISCUSSION THREAD

**Discussion by tigerinus** [pyinstaller/pyinstaller#8285] [Help] [answered]
After packaged, we'd like to install the built binary under `/usr/local/bin` - but we would also have to put `_internal/` folder under the same path.

Is there a way to specify a different location, e.g. `/usr/local/lib`, instead of `/usr/local/bin/_internal`?

Thanks.

**Comment by rokm** [comment, marked as answer]
The name of the contents directory is determined at build time, and can be controlled via [`--contents-directory` command-line option](https://pyinstaller.org/en/stable/usage.html#cmdoption-contents-directory) (if you're building directory from source), or by adding `contents_directory='name'` argument to `EXE` in the .spec file (if you're building from spec).

But FWIW, application bundles like that are usually not placed directly into `/usr/bin` or `/usr/local/bin`; instead, one would place them in some other location (for example, `/opt/my-program`), and then symlink the executable (`/opt/my-program/my-program`) to `/usr/local/bin`.
